jlm/hls/ir/hls.cpp

```diff
@@ -135,28 +142,26 @@ loop_node::set_predicate(jlm::rvsdg::output * p)
 }
 
 std::shared_ptr<const bundletype>
-get_mem_req_type(const rvsdg::valuetype & elementType, bool write)
+get_mem_req_type(std::shared_ptr<const rvsdg::valuetype> elementType, bool write)
 {
-  auto elements =
-      new std::vector<std::pair<std::string, std::shared_ptr<const jlm::rvsdg::type>>>();
-  elements->emplace_back("addr", llvm::PointerType::Create());
-  elements->emplace_back("size", std::make_unique<jlm::rvsdg::bittype>(4));
-  elements->emplace_back("id", std::make_unique<jlm::rvsdg::bittype>(8));
+  std::vector<std::pair<std::string, std::shared_ptr<const jlm::rvsdg::type>>> elements;
+  elements.emplace_back("addr", llvm::PointerType::Create());
+  elements.emplace_back("size", std::make_unique<jlm::rvsdg::bittype>(4));
+  elements.emplace_back("id", std::make_unique<jlm::rvsdg::bittype>(8));
```

Comment:
Should this not be `rvsdg::bittype::Create(4)` and `rvsdg::bittype::Create(8)`, respectively?

Reply:
good catch! I find several of these, but missed this one